core/commands/storage.go

```diff
@@ -725,23 +725,32 @@ func solveChallenge(chunkInfo *storage.Chunk, chunkHash string, ssID string, res
 	sm := storage.GlobalSession
 	ss := sm.GetOrDefault(ssID)
 
+	// get challenge object from params
 	r := ChallengeRes{}
 	if err := json.Unmarshal(resBytes, &r); err != nil {
 		log.Error(err)
 		sendSessionStatusChan(ss.SessionStatusChan, storage.UploadStatus, false, err)
 		storage.GlobalSession.Remove(ssID, chunkHash)
+		return
+	}
+	// find chunk hash cid
+	chunkCid, err := cidlib.Parse(chunkHash)
```

Comment:
Yes, bad naming because this code path was written before Q4 planning. Shelly is going to work on the naming. Think of chunkInfo as shardInfo here. In order to not overcomplicate changes, the renaming will be done in a future ticket soon. I changed this so that btfs would compile.